workitem/expression_compiler.go

```diff
@@ -425,6 +425,65 @@ func (c *expressionCompiler) Not(e *criteria.NotExpression) interface{} {
 	return c.binary(e, "!=")
 }
 
+func (c *expressionCompiler) Child(e *criteria.ChildExpression) interface{} {
+	left, ok := e.Left().(*criteria.FieldExpression)
+	if !ok {
+		c.err = append(c.err, errs.Errorf("invalid left expression (not a field expression): %+v", e.Left()))
+		return nil
+	}
+
+	if strings.Contains(left.FieldName, "'") {
+		// beware of injection, it's a reasonable restriction for field names,
+		// make sure it's not allowed when creating wi types
+		c.err = append(c.err, errs.Errorf("single quote not allowed in field name: %s", left.FieldName))
+		return nil
+	}
+	litExp, ok := e.Right().(*criteria.LiteralExpression)
+	if !ok {
+		c.err = append(c.err, errs.Errorf("failed to convert right expression to literal expression: %+v", e.Right()))
+		return nil
+	}
+	r, ok := litExp.Value.(string)
+	if !ok {
+		c.err = append(c.err, errs.Errorf("failed to convert value of right literal expression to string: %+v", litExp.Value))
+		return nil
+	}
+	if strings.Contains(r, "'") {
+		// beware of injection, it's a reasonable restriction for field names,
+		// make sure it's not allowed when creating wi types
+		c.err = append(c.err, errs.Errorf("single quote not allowed in field value: %s", r))
+		return nil
+	}
+
+	var tblName string
+	var tblAlias string
+	var tblJoin string
+	if left.FieldName == SystemIteration {
+		tblName = "iterations"
+		tblAlias = "iter"
+		tblJoin = "iteration"
+	} else if left.FieldName == SystemArea {
+		tblName = "areas"
+		tblAlias = "ar"
+		tblJoin = "area"
+	} else {
+		c.err = append(c.err, errs.Errorf("invalid field name: %+v", left.FieldName))
+		return nil
+	}
+	c.joins[tblJoin].Active = true
+	c.parameters = append(c.parameters, r)
+
+	return fmt.Sprintf(`(uuid("work_items".fields->>'%[1]s') IN (
```

Comment:
Please leave a comment here what this query is doing. It needs explanation. You could use `WorkItemStorage{}.TableName{}` instead of `work_items` but it is unlikely to change.

Reply:
Done. 80f71030e5e692c85ff1a6109a7a15474de1bc9f